feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/domain/validation/validations/SwapFeeSufficientBalanceValidation.kt

```diff
@@ -0,0 +1,46 @@
+package io.novafoundation.nova.feature_swap_impl.domain.validation.validations
+
+import io.novafoundation.nova.common.utils.isZero
+import io.novafoundation.nova.common.validation.ValidationStatus
+import io.novafoundation.nova.common.validation.valid
+import io.novafoundation.nova.common.validation.validationError
+import io.novafoundation.nova.feature_swap_impl.domain.validation.SwapValidation
+import io.novafoundation.nova.feature_swap_impl.domain.validation.SwapValidationFailure
+import io.novafoundation.nova.feature_swap_impl.domain.validation.SwapValidationFailure.InsufficientBalance
+import io.novafoundation.nova.feature_swap_impl.domain.validation.SwapValidationPayload
+import io.novafoundation.nova.feature_swap_impl.domain.validation.maxAmountToSwap
+import io.novafoundation.nova.feature_swap_impl.domain.validation.swapAmountInFeeToken
+import io.novafoundation.nova.feature_swap_impl.domain.validation.toBuyAmountToKeepEDInFeeAsset
+import io.novafoundation.nova.feature_swap_impl.domain.validation.totalDeductedAmountInFeeToken
+import java.math.BigInteger
+
+class SwapFeeSufficientBalanceValidation : SwapValidation {
+
+    override suspend fun validate(value: SwapValidationPayload): ValidationStatus<SwapValidationFailure> {
+        val swapAmountInFeeToken = value.swapAmountInFeeToken
+        val totalDeductedAmount = value.totalDeductedAmountInFeeToken
+
+        if (value.feeAsset.transferableInPlanks < swapAmountInFeeToken + totalDeductedAmount) {
+            val chainAssetIn = value.detailedAssetIn.asset.token.configuration
+            val chainAssetOut = value.detailedAssetOut.asset.token.configuration
+            val feeAsset = value.feeAsset.token.configuration
+            val maxAmountToSwap = value.maxAmountToSwap
+            val toBuyAmountToKeepEDInFeeAsset = value.toBuyAmountToKeepEDInFeeAsset
+
+            return if (toBuyAmountToKeepEDInFeeAsset.isZero) {
+                InsufficientBalance.NoNeedsToBuyMainAssetED(chainAssetIn, feeAsset, value.maxAmountToSwap, value.swapFee.networkFee).validationError()
+            } else {
+                InsufficientBalance.NeedsToBuyMainAssetED(
+                    value.feeAsset.token.configuration,
+                    chainAssetIn,
+                    toBuyAmountToKeepEDInCommissionAsset = toBuyAmountToKeepEDInFeeAsset,
+                    toSellAmountToKeepEDUsingAssetIn = BigInteger.ZERO, // TODO how to convert toBuyAmountToKeepEDInFeeAsset to this value?
```

Comment:
Let me know please the nice way to convert toBuyEd amount (commission asset) to toSellAmount (assetIn) with sufficient accuracy.

Reply:
You should use `swapFee.minimumBalanceBuyIn.commissionAssetToSpendOnBuyIn`, it is a sealed class that will be a MinimumBalanceBuyIn.NeedsToBuyMinimumBalance`, which have all required info you need here